processor/src/queues.ts

```diff
@@ -180,7 +181,14 @@ async function sync(gitDir, checkoutDir) {
     .acquire(gitDir, async done => {
       log.info('Acquired sync lock for ', gitDir)
 
+      const registryHash = getRegistryHash(gitDir)
+
       if (await exists(checkoutDir)) {
+        if (registryHash) {
+          // no need to pull if we aren't going to use the latest commit
+          done()
+        }
```

Comment:
But you are still pulling below, you are just releasing the lock early?

Reply:
Resolved in 68230b6.